src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1178,45 +1165,29 @@ public PortModel AddPort(PortType portType, PortData data, int index)
             switch (portType)
             {
                 case PortType.Input:
-
-                    bool hasDefaultValue = data.DefaultValue != null;
-
                     if (inPorts.Count > index)
                     {
                         p = inPorts[index];
-
-                        //update the name on the node
-                        //e.x. when the node is being re-registered during a custom
-                        //node save
-                        p.PortName = data.NickName;
-                        if (hasDefaultValue)
-                        {
-                            p.UsingDefaultValue = true;
-                            p.DefaultValueEnabled = true;
-                        }
-                        p.ToolTipContent = data.ToolTipString;
-                        return p;
+                        p.SetPortData(data);
```

Comment:
As `PortName`, `DefaultValueEnabled`, `ToolTipContent` are all directly from `PortData`, instead of copying them, `PortModel` has a reference to `PortData`. Here I don't create a `Data` property because we are not interested in `PortData` in `PortModel`, but only interested in tooltip, nickname and so on.